File: LISTA-02(JAVA)/Q2.java
class Viagem {
    private double d = 0;
    private double t = 0;

    public void setDistancia(double v) {
        if (v >= 0) this.d = v;
        else throw new IllegalArgumentException("Valor negativo é inválido");
    }
        public void setTempo(double v) {
        if (v >= 0) this.t = v;
        else throw new IllegalArgumentException("Valor negativo é inválido");
    }
    public double getDistancia() {
        return this.d;
    }
    public double getTempo() {
        return this.t;
    }

    public double calcVel() {
        return this.d / this.t;
    }
}

public class Q2 {
    public static void main(String[] args) {
        Viagem x = new Viagem();
        x.setDistancia(20);
        x.setTempo(10);
        System.out.println("A velocidade média é: " + x.calcVel());
    }
}

File: LISTA-02(PYTHON)/Q2.py
class Viagem :
    def __init__(self) :
        self.__d = 0
        self.__t = 0
    def set_distancia(self, v) :
        if v >= 0 :
            self.__d = v
        else :
            raise ValueError("Distância Inválida")
    def set_tempo(self, v) :
        if v >= 0 :
            self.__t = v
        else :
            raise ValueError("Tempo Inválido")
    def get_distancia(self) :
        return self.__d
    def get_tempo(self) :
        return self.__t
    def calc_vel_media(self) :
        vel_media = self.__d / self.__t
        return vel_media
    
class UI :
    @staticmethod
    def main() :
         x = Viagem()
         x.set_distancia(100)
         x.set_tempo(1)
         print(x.calc_vel_media())

UI.main()